File: pathplannerlib/src/main/java/com/pathplanner/lib/auto/EventManager.java
package com.pathplanner.lib.auto;

import com.pathplanner.lib.commands.WrappedEventCommand;
import edu.wpi.first.math.Pair;
import edu.wpi.first.wpilibj2.command.*;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class EventManager {
  private static final HashMap<String, Command> eventMap = new HashMap<>();

  public static void registerCommand(String name, Command command) {
    eventMap.put(name, command);
  }

  public static void registerCommands(List<Pair<String, Command>> commands) {
    for (var pair : commands) {
      registerCommand(pair.getFirst(), pair.getSecond());
    }
  }

  public static void registerCommands(Map<String, Command> commands) {
    eventMap.putAll(commands);
  }

  public static boolean hasCommand(String name) {
    return eventMap.containsKey(name);
  }

  public static Command getCommand(String name) {
    if (hasCommand(name)) {
      return new WrappedEventCommand(eventMap.get(name));
    } else {
      return Commands.none();
    }
  }
}

File: pathplannerlib/src/main/native/cpp/pathplanner/lib/auto/EventManager.cpp
#include "pathplanner/lib/auto/EventManager.h"
#include "pathplanner/lib/auto/CommandUtil.h"

using namespace pathplanner;

std::unordered_map<std::string, std::shared_ptr<frc2::Command>> EventManager::eventMap =
		std::unordered_map<std::string, std::shared_ptr<frc2::Command>>();

void EventManager::registerCommand(std::string name,
		std::shared_ptr<frc2::Command> command) {
	EventManager::eventMap.emplace(name, command);
}

bool EventManager::hasCommand(std::string name) {
	return EventManager::eventMap.find(name) != EventManager::eventMap.end();
}

frc2::CommandPtr EventManager::getCommand(std::string name) {
	if (EventManager::hasCommand(name)) {
		return CommandUtil::wrappedEventCommand(EventManager::eventMap.at(name));
	}
	return frc2::cmd::None();
}
